### ISSUE TITLE
pandas' <NA> becomes either NULL or NaN when inserted into ClickHouse table

### ISSUE THREAD

**Issue by konst-ivanov** [ClickHouse/clickhouse-connect#412]
### Describe the bug
pandas' `<NA>` turns to
- `NULL` when inserted into `Nullable(UIntX)` or `Nullable(IntX)` column
- `NaN` when inserted into `Nullable(FloatX)`

### Steps to reproduce
1. run clickhouse server
2. run code example
3. check what's been inserted into ClickHouse table

### Expected behaviour
`<NA>` becoming the same value for both Int and Float nullable columns (I'd prefer `NULL`)

### Code example
```python
import pandas as pd
from clickhouse_connect import get_client


def main():
    client = get_client(host="localhost", database="default")
    client.command("DROP TABLE IF EXISTS insert_df_test")
    client.command(
        """CREATE TABLE insert_df_test
        (
            row_id UInt64,
            uint64_col Nullable(UInt64),
            int64_col Nullable(Int64),
            float64_col Nullable(Float64)
        )
        ENGINE MergeTree
        ORDER BY row_id
        """
    )
    df = pd.DataFrame(
        [[0, 1, -1, 1.0], [1, None, None, None]],
        columns=["row_id", "uint64_col", "int64_col", "float64_col"],
    )
    df.uint64_col = df.uint64_col.astype("Int64")
    df.int64_col = df.int64_col.astype("Int64")
    df.float64_col = df.float64_col.astype("Float64")
    client.insert_df("insert_df_test", df)


if __name__ == '__main__':
    main()
```

#### Environment
* clickhouse-connect version: 0.8.3
* Python version: 3.11.8
* Operating system: Win/MacOs

#### ClickHouse server
* ClickHouse Server version: 24.9.2.42

**Comment by genzgd** [comment]
This is expected behavior.  NaN is a valid float in Pandas and ClickHouse, so it would be breaking changing to insert NULL instead.  If you want to actually insert a null, you can convert your pandas column to an `object` column and replace NaN with `None` before inserting the dataframe.

**Comment by genzgd** [comment]
In fact, your code is probably doing the conversion, not `clickhouse_connect`:

```python
 df.float64_col = df.float64_col.astype("Float64")
```

**Comment by genzgd** [comment]
Okay, I was wrong.  This seems to be related to recent Pandas types.  Let me see if there's a way to determine the difference between <NA> and NaN

**Comment by genzgd** [comment]
This should be fixed in 0.8.4.  Pandas apparently just treats np.nan/float NaN as an <NA> regardless
